webui/react/src/components/InteractiveTable.tsx

```diff
@@ -0,0 +1,510 @@
+  onResizeStop: DraggableEventHandler;
+  title: unknown;
+  width: number;
+}
+
+interface CellProps {
+  children?: React.ReactNode;
+  className: string;
+  isCellRightClickable?: boolean;
+}
+
+export const PAGE_PADDING = 16;
+
+const RightClickableRowContext = createContext({});
+
+const Row = ({
+  className,
+  children,
+  record,
+  ContextMenu,
+  areRowsSelected,
+  ...props
+}: RowProps) => {
+  const classes = [ className, css.row ];
+
+  const [ rowHovered, setRowHovered ] = useState(false);
+  const [ rightClickableCellHovered, setRightClickableCellHovered ] = useState(false);
+  const [ contextMenuOpened, setContextMenuOpened ] = useState(false);
+
+  if (areRowsSelected) {
+    return <tr className={classes.join(' ')} {...props}>{children}</tr>;
+  }
+
+  const rightClickableCellProps = {
+    onContextMenu: (e : React.MouseEvent) => e.stopPropagation(),
+    onMouseEnter: () => setRightClickableCellHovered(true),
+    onMouseLeave: () => setRightClickableCellHovered(false),
+  };
+
+  const rowContextMenuTriggerableOrOpen =
+    (rowHovered && !rightClickableCellHovered) || contextMenuOpened;
+
+  if (rowContextMenuTriggerableOrOpen) {
+    classes.push('ant-table-row-selected');
+  }
+  return record ? (
+    <RightClickableRowContext.Provider value={{ ...rightClickableCellProps }}>
+      <ContextMenu record={record} onVisibleChange={setContextMenuOpened}>
+        <tr
+          className={classes.join(' ')}
+          onMouseEnter={() => setRowHovered(true)}
+          onMouseLeave={() => setRowHovered(false)}
+          {...props}>
+          {children}
+        </tr>
+      </ContextMenu>
+    </RightClickableRowContext.Provider>
+  ) : (
+    <tr className={classes.join(' ')} {...props}>{children}</tr>
+  );
+};
+
+const Cell = ({ children, className, isCellRightClickable, ...props }: CellProps) => {
+  const rightClickableCellProps = useContext(RightClickableRowContext);
+  const classes = [ className, css.cell ];
+  if (!isCellRightClickable) return <td className={classes.join(' ')} {...props}>{children}</td>;
+  return (
+    <td className={classes.join(' ')} {...props}>
+      <div className={css.rightClickableCellWrapper} {...rightClickableCellProps}>
+        {children}
+      </div>
+    </td>
+  );
+};
+
+const HeaderCell = ({
+  onResize,
+  onResizeStart,
+  onResizeStop,
+  width,
+  className,
+  columnName,
+  filterActive,
+  moveColumn,
+  index,
+  title: unusedTitleFromAntd,
+  isResizing,
+  dropRightStyle,
+  dropLeftStyle,
+  dragState,
+  ...props
+}: HeaderCellProps) => {
+  const resizingRef = useRef<HTMLDivElement>(null);
+
+  const headerCellClasses = [ css.headerCell ];
+  const dropTargetClasses = [ css.dropTarget ];
+
+  const [ , drag ] = useDrag({
+    canDrag: () => !isResizing,
+    item: { index },
+    type,
+  });
+
+  const [ { isOver }, drop ] = useDrop({
+    accept: type,
+    canDrop: (_, monitor) => {
+      const dragItem = (monitor.getItem() || {}); // as DndItem;
```

Comment:
nit: either uncomment the type assertion or remove it

Reply:
done.